lib/puppet/module/task.rb

```diff
@@ -61,6 +62,41 @@ def self.is_tasks_filename?(path)
       return true
     end
 
+    # Find task's required lib files and retrieve paths
+    # for both 'files' and 'implementation:files' metadata keys
+    def self.find_files(files, mod)
+      env = mod.environment.name
+      # raise EnvironmentNotFound if applicable
+      Puppet.lookup(:environments).get!(env)
```

Comment:
Could this ever happen? Not sure how it would be added to the module first.

Reply:
I think this gets called when we instantiate the module to begin with. So...probably don't need the check twice?